src/main.sh

```diff
@@ -4,9 +4,12 @@ set -e
 
 function _check_setup {
   # Checking git and secret-plugin setup:
-  if [[ ! -d ".git" ]] || [[ ! -d ".git/hooks" ]]; then
+  set +e
+  git rev-parse --is-inside-work-tree >/dev/null 2>&1
```

Comment:
It is better to move this function into `utils`. I guess it could be named `_is_inside_git_tree`.
And call it here like `local is_tree = $(_is_inside_git_tree)`
So, the tasks are:

- [ ] Move this function into `utils` file and `git` section
- [ ] Test it, both scenarios: inside and outside